src/main/java/de/tum/cit/aet/artemis/atlas/service/AtlasAgentToolsService.java

```diff
@@ -0,0 +1,224 @@
+            @ToolParam(description = "the taxonomy level (REMEMBER, UNDERSTAND, APPLY, ANALYZE, EVALUATE, CREATE)") String taxonomyLevel) {
+        try {
+            Optional<Course> courseOpt = courseRepository.findById(courseId);
+            if (courseOpt.isEmpty()) {
+                return toJson(Map.of("error", "Course not found with ID: " + courseId));
+            }
+
+            Course course = courseOpt.get();
+            Competency competency = new Competency();
+            competency.setTitle(title);
+            competency.setDescription(description);
+            competency.setCourse(course);
+
+            if (taxonomyLevel != null && !taxonomyLevel.isEmpty()) {
+                try {
+                    CompetencyTaxonomy taxonomy = CompetencyTaxonomy.valueOf(taxonomyLevel.toUpperCase());
+                    competency.setTaxonomy(taxonomy);
+                }
+                catch (IllegalArgumentException e) {
+                    log.warn("Invalid taxonomy level '{}', using default", taxonomyLevel);
+                }
+            }
+
+            Competency savedCompetency = competencyRepository.save(competency);
+
+            Map<String, Object> competencyData = new LinkedHashMap<>();
+            competencyData.put("id", savedCompetency.getId());
+            competencyData.put("title", savedCompetency.getTitle());
+            competencyData.put("description", savedCompetency.getDescription());
+            competencyData.put("taxonomy", savedCompetency.getTaxonomy() != null ? savedCompetency.getTaxonomy().toString() : "");
+            competencyData.put("courseId", courseId);
+
+            Map<String, Object> response = new LinkedHashMap<>();
+            response.put("success", true);
+            response.put("competency", competencyData);
+
+            return toJson(response);
+        }
+        catch (Exception e) {
+            log.error("Error creating competency for course {}: {}", courseId, e.getMessage(), e);
+            return toJson(Map.of("error", "Failed to create competency: " + e.getMessage()));
+        }
+    }
+
+    /**
+     * Tool for getting course description.
+     *
+     * @param courseId the course ID
+     * @return the course description or empty string if not found
+     */
+    @Tool(description = "Get the description of a course")
+    public String getCourseDescription(@ToolParam(description = "the ID of the course") Long courseId) {
+        try {
+            Optional<Course> courseOpt = courseRepository.findById(courseId);
+            if (courseOpt.isEmpty()) {
+                return "";
+            }
+
+            Course course = courseOpt.get();
+            String description = course.getDescription();
+            return description != null ? description : "";
+        }
+        catch (Exception e) {
+            log.error("Error getting course description for course {}: {}", courseId, e.getMessage(), e);
+            return "";
+        }
```

Comment:
```suggestion
        return courseRepository.findById(courseId).map(Course::getDescription).orElse("");
```
Could be written much more concisely.